components/calendar/calendar.js

```diff
@@ -467,20 +467,21 @@ class Calendar extends LocalizeCoreElement(RtlMixin(LitElement)) {
 				// role="gridcell" used for screen reader (e.g., JAWS and VoiceOver) behavior to work properly
 				return html`
 					<td
+						aria-label="${description}"
 						aria-selected="${selected ? 'true' : 'false'}"
 						data-date=${date}
 						data-month=${month}
 						data-year=${year}
 						@keydown="${this._onKeyDown}"
-						role="gridcell"
```

Comment:
I see that the `table` is being marked up with `role="presentation"`. When I inspect the "rows" and the "cells" in the Accessibility Object Model viewer in Chrome, I notice that the `tr`s don't have a role at all, and the `td`s have a generic role unless one is explicitly specified. Whereas when `role="presentation"` is removed (normal table semantics), the `tr` and `td` elements get their default `row` and `gridcell` roles applied automatically as one would expect. 

So, with your current changes, the `td` is getting a generic role assigned to it (at least in Chrome on MacOS), which I think makes sense since we're basically telling the browser to ignore table semantics. If on the other hand we actually want grid semantics, we either need to be using the table elements with their default roles, or we need to be assigning the appropriate `row` and `gridcell` roles. Also worth noting that Mozilla states that `gridcell` must be contained within a `row`, so what we were specifying previously was technically invalid.

Not sure if that clears anything up with respect to the roles. Currently we are using the table element simply for layout. We're not leveraging any of it's semantics, at least not properly, and the cell announcement comes entirely from the label. The header cells are purely for sighted users. I can't help but wonder if some of the crazy inconsistency could be avoided if we were just using Aria (if we actually want grid semantics) + CSS grid (for layout) instead of table elements.